### PR TITLE
Change NuGet cache to be in repository root

### PR DESCRIPTION
**Pull request by Youssef1313**
Related to https://github.com/dotnet/msbuild/pull/7187.

### CODE REVIEW

**Comment by Youssef1313** [comment]
@elachlan This seems to work for CI. But I'm not sure if it will work well for local development. Haven't cloned the repository so I can't test.

File: .editorconfig [hunk 1]
```diff
@@ -202,3 +202,7 @@ end_of_line = crlf
 [src/**/*.{cs,vb}]
 # IDE0005: Remove unnecessary usings/imports
 dotnet_diagnostic.IDE0005.severity = warning
+
+[.nuget/**/*.cs]
+# Instantiate argument exceptions correctly
+dotnet_diagnostic.CA2208.severity = suggestion
```
Comment (elachlan) [inline .editorconfig:208]: Should this be `dotnet_diagnostic.severity = suggestion`?

Comment (Youssef1313) [reply .editorconfig:208]: I'm disabling CA2208 specifically so that you can clearly see what rules are violated by the referenced packages.

**Comment by elachlan** [comment]
@Youssef1313 This doesn't work locally. I still get the standard errors. Thanks for your efforts though.

```
\.nuget\packages\microsoft.codeanalysis.collections\4.0.0-4.21379.20\contentFiles\cs\nets
tandard2.0\ImmutableSegmentedList`1.cs(346,23): error CA2208: Call the ArgumentException constructor that contains a me
ssage and/or paramName parameter [D:\Development\msbuild\src\Framework\Microsoft.Build.Framework.csproj]
```

@Forgind I think this solution might work if we can fix that quirk with local build. Otherwise we can just move back to using editorconfigs.

**Comment by Youssef1313** [comment]
@elachlan Do you have `NUGET_PACKAGES` environment variable set to anything?

**Comment by elachlan** [comment]
> @elachlan Do you have `NUGET_PACKAGES` environment variable set to anything?

I just run "build" from the msbuild root via cmd prompt. As far as I can tell that env variable is not set.

**Comment by Youssef1313** [comment]
> > @elachlan Do you have `NUGET_PACKAGES` environment variable set to anything?
> 
> I just run "build" from the msbuild root via cmd prompt. As far as I can tell that env variable is not set.

The environment variable is the only thing that can take precedence over NuGet.config. I think arcade has set it for your local environment?

https://github.com/dotnet/msbuild/blob/90275e45541681fa48c1acc0fffe4bd23821ce43/eng/common/tools.ps1#L49

Here, `useGlobalNuGetCache` is likely set to `!ci` which is `true` for your local environment. Then:

https://github.com/dotnet/msbuild/blob/90275e45541681fa48c1acc0fffe4bd23821ce43/eng/common/tools.ps1#L608-L623

**Comment by Youssef1313** [comment]
I think we can just pass false for `useGlobalNuGetCache` from the build script and things should work smoothly?

**Comment by elachlan** [comment]
We need to check with the team to confirm this is a good idea. They might dislike the idea of not using the global cache.

It might be simpler to just use the editorconfig in conjunction with the globalconfig as outlined in https://github.com/dotnet/roslyn/issues/55992

**Comment by Youssef1313** [comment]
Sure. I'm going to close for now. You can incorporate either using editorconfigs and globalconfigs, or using the local nuget cache approach in #7187 as you and the team see what's more suitable.

**Comment by elachlan** [comment]
> Sure. I'm going to close for now. You can incorporate either using editorconfigs and globalconfigs, or using the local nuget cache approach in #7187 as you and the team see what's more suitable.

Thank you so much for your help on this. It gives us a good idea on what is possible.

**Pull request closed without merging**